Example/Shared/WiFi/ProvisionOverWiFiView+VM.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  ProvisionOverWiFiView+VM.swift
+//  nRF-Wi-Fi-Provisioner
+//
+//  Created by Nick Kibysh on 20/02/2024.
+//
+
+import SwiftUI
+import NordicWiFiProvisioner_SoftAP
+
+extension ProvisionOverWiFiView {
+    @MainActor
+    class ViewModel: ObservableObject {
+        enum LedStatus {
+            case notDetermined, on, off, error(Error)
+            
+            var disabled: Bool {
+                switch self {
+                case .on, .off: return false
+                default: return true
+                }
+            }
+        }
+        
+        enum Status {
+            case notConnected
+            case connecting
+            case connected
+            case error(_ error: Error)
+        }
+        
+        @Published private (set) var status = Status.notConnected
+        private var manager = ProvisionManager()
+        @Published private (set) var ssids: [ReportedSSID] = []
+        @Published var selectedSSID: ReportedSSID?
+        @Published var ssidPassword: String = ""
+        
+        @Published private (set) var led1Status = LedStatus.notDetermined
+        @Published private (set) var led2Status = LedStatus.notDetermined
+        
+        @Published private (set) var alertError: TitleMessageError? = nil
```

Comment:
There's no need to show a screen with an error message. Alert is enough usually.